spec/services/pairing_strategies/single_sided_swiss_spec.rb

```diff
@@ -355,26 +355,100 @@
+          round1.pairings.delete_all
+          create(:pairing, round: round1, player1: snap, player2: crackle, score1: 0, score2: 3,
+                           side: :player1_is_runner)
+          create(:pairing, round: round1, player1: pop, player2: nil, score1: 1)
 
+          # Set up Round 2
+          round2.completed = true
+          round2.save
+          create(:pairing, round: round2, player1: snap, player2: crackle, score1: 3, score2: 0,
+                           side: :player1_is_corp)
+          create(:pairing, round: round2, player1: pop, player2: nil, score1: 1)
+
+          # Artificial Standings
+          # Snap    3
+          # Crackle 3
+          # Pop     2
+
+          round3 = create(:round, tournament:, stage:, number: 3)
+          pairer = described_class.new(round3)
           pairer.pair!
 
-          round2.reload
-          round2.pairings.each do |pairing|
-            expect(pairing.players).not_to include(pop) if pairing.players.include? snap
+          round3.reload
+          round3.pairings.each do |pairing|
+            expect(pairing.players).not_to include(snap) if pairing.players.include? crackle
           end
         end
 
-        it 'avoids second bye' do
-          create(:pairing, player1: crackle, player2: nil)
+        it 'avoids second bye when real pairing was a tie' do
+          # Round 1:
+          #   Snap (1) vs Crackle (1) - tie so neither player gets the bye only on points.
+          #   Pop (bye)
+          # Standings
+          #   Pop     3 (bye)
+          #   Snap    1
+          #   Crackle 1
+          # Round 2:
+          #   Pop (3) vs Crackle|Snap
+          #   Crackle|Snap: Bye
+
+          round1.pairings.delete_all
+          create(:pairing, round: round1, player1: snap, player2: crackle, score1: 1, score2: 1,
+                           side: :player1_is_runner)
+          create(:pairing, round: round1, player1: pop, player2: nil, score1: 3)
 
           pairer.pair!
 
-          round2.reload
           round2.pairings.each do |pairing|
-            expect(pairing.players).not_to include(crackle) if pairing.players.include? nil_player
+            expect(pairing.players).not_to include(pop) if pairing.players.include? nil_player
+          end
+        end
+
+        it 'avoids second bye really in the third round.' do
+          # Round 1:
+          #   Snap (3) vs Crackle (0)
+          #   Pop (bye)
+          # Standings
+          #   Snap    3
+          #   Pop     3 (bye)
+          #   Crackle 0
+          # Round 2:
+          #   Snap (3) vs Pop (0)
+          #   Crackle: Bye
+          # Standings
+          #   Snap (6)
+          #   Pop (3)
+          #   Crackle (3)
+          # Round 3:
+          #   Pop vs Crackle
+          #   Snap (Bye)
+
+          # expect(pairing.side_for(snap)).to eq(:runner) if pairing.players.include? snap
+          # expect(pairing.side_for(crackle)).to be_nil if pairing.players.include? crackle
```

Comment:
Was it intentional to leave these two lines in?

Reply:
nope